File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> mp;
        unordered_map<char, int> tp;
        if (t.size() != s.size())
            return false;
        for (int i = 0; i < s.size(); i++) {
            mp[s[i]]++;
        }
        for (int j = 0; j < t.size(); j++) {
            tp[t[j]]++;
        }
        for (int i = 0; i < s.size(); i++) {
            if (mp[s[i]] != tp[s[i]])
                return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        HashMap<Character, Integer> mp = new HashMap<>();
        HashMap<Character, Integer> tp = new HashMap<>();
        for (char c : s.toCharArray()) {
            mp.put(c, mp.getOrDefault(c, 0) + 1);
        }
        for (char c : t.toCharArray()) {
            tp.put(c, tp.getOrDefault(c, 0) + 1);
        }
        return mp.equals(tp);

    }
}